pywbem/_subscription_manager.py

```diff
@@ -1284,16 +1301,60 @@ def _create_filter(self, server_id, source_namespace, query,
                         filter_inst = server.conn.GetInstance(filter_path)
                         self._owned_filters[server_id][i] = filter_inst
                     return filter_inst
-            filter_path = server.conn.CreateInstance(filter_inst)
+
+            filter_path = server.conn.CreateInstance(
+                filter_inst, namespace=server.interop_ns)
             filter_inst = server.conn.GetInstance(filter_path)
             self._owned_filters[server_id].append(filter_inst)
         else:
             # Responsibility to ensure it does not exist yet is with the user
-            filter_path = server.conn.CreateInstance(filter_inst)
+            filter_path = server.conn.CreateInstance(
+                filter_inst, namespace=server.interop_ns)
             filter_inst = server.conn.GetInstance(filter_path)
 
+        # If systemname does not yet exist, create it
+        if server_id not in self._systemnames:
+            self._systemnames[server_id] = filter_inst['SystemName']
+        else:
+            assert self._systemnames[server_id] == filter_inst['SystemName']
+
         return filter_inst
 
+    def _get_systemname(self, server_id):
+        """
+        Get the SystemName returned for anything returned for this server_id.
+        This method gets the s
```

Comment:
Incomplete

Reply:
Added.

Reply:
DONE

Reply:
Whatever you did, it does not show up in the PR update.

Reply:
outdated - function was removed.